src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -76,7 +76,7 @@ static bool genIsTableDrivenHWIntrinsic(NamedIntrinsic intrinsicId, HWIntrinsicC
     // TODO - make more categories to the table-driven framework
     // HW_Category_Helper and HW_Flag_SpecialCodeGen usually need manual codegen
     const bool tableDrivenCategory =
-        (category != HW_Category_Special) && (category != HW_Category_Scalar) && (category != HW_Category_Helper);
+        (category != HW_Category_Special) && (category != HW_Category_Helper);
```

Comment:
The bigger refactoring is here in `hwintrinsiccodegenxarch.cpp`. It namely moves some logic around to ensure we only have one place that is doing the `ConsumeRegs`/`ProduceReg` which allows all paths to share the embedded broadcast/masking handling and centralizes instruction selection.

Reply:
Most of this is just moving code around, renaming parameters for clarity, or passing down the computed instruction to avoid duplicated work and handling.